### ISSUE TITLE
Reacharound block placing

### ISSUE THREAD

**Issue by careeoki** [PixelGuys/Cubyz#727]
Place blocks on faces not visible to the player.
![tangiblemediumandeancat-size_restricted](https://github.com/user-attachments/assets/bdc543ed-f662-447d-b601-0e2a64ce5ae4)
The selected face should be highlighted in some way to show where the block will be placed.

**Comment by IntegratedQuantum** [comment]
I do agree that it would be nice to be able to place blocks behind/below other blocks. Main use-cases I think would be speed bridging and extending a building that's in the air downwards (basically skyblock).

However I don't think this is the best solution, it seems like you need to be very precise and there would be edge cases where it's unclear where it whether it should prefer placing on the distant block or behind the near block.

I would propose to have a keybind (or maybe a tool that you put in your off-hand?) that lets you switch between different placement modes, one being normal placement, and the other being a sort of behind placement, where it shows you even the invisible bit of the selection box and your block gets placed on the back of it.

**Comment by careeoki** [comment]
Yeah, this gif is bad, I just used it to show what I was talking about. 
I think this is something the player should have at all times, and doesn't need a tool to do it
